File: Desktop/optirh/src/main/java/tn/nexus/Entities/Mission/Mission.java
package tn.nexus.Entities.Mission;

import java.sql.Timestamp;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class Mission {
    private int id;
    private String titre;
    private String description;
    private String status; // To Do, In Progress, Done
    private int projectId; // Référence au projet
    private int assignedTo; // Référence à l'utilisateur assigné
    private Timestamp createdAt;
    private Timestamp updatedAt;
    private Timestamp dateTerminer; // Nouvel attribut

    public Mission(int id, String titre, String description, String status, int projectId, int assignedTo, Timestamp createdAt, Timestamp updatedAt, Timestamp dateTerminer) {
        this.id = id;
        this.titre = titre;
        this.description = description;
        this.status = status;
        this.projectId = projectId;
        this.assignedTo = assignedTo;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
        this.dateTerminer = dateTerminer;
    }
    // Constructeurs
    public Mission() {}

    public Mission(int id, String titre, String description, String status, int projectId, int assignedTo, Timestamp createdAt, Timestamp updatedAt) {
        this.id = id;
        this.titre = titre;
        this.description = description;
        this.status = status;
        this.projectId = projectId;
        this.assignedTo = assignedTo;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }

    public Mission(LocalDate createdAt, LocalDate dateTerminer, String s) {

    }

    // Getters et Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitre() {
        return titre;
    }

    public void setTitre(String titre) {
        this.titre = titre;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public int getProjectId() {
        return projectId;
    }

    public void setProjectId(int projectId) {
        this.projectId = projectId;
    }

    public int getAssignedTo() {
        return assignedTo;
    }

    public void setAssignedTo(int assignedTo) {
        this.assignedTo = assignedTo;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }

    public Timestamp getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Timestamp updatedAt) {
        this.updatedAt = updatedAt;
    }

    @Override
    public String toString() {
        return "Mission{" +
                "id=" + id +
                ", titre='" + titre + '\'' +
                ", status='" + status + '\'' +
                '}';
    }

    public Timestamp getDateTerminer() { return dateTerminer; }
    public void setDateTerminer(Timestamp dateTerminer) { this.dateTerminer = dateTerminer; }
}

File: WebTemplate/Adminto/src/Entity/GsProjet/Mission.php
<?php

namespace App\Entity\GsProjet;

use App\Repository\GsProjet\MissionRepository;
use Doctrine\ORM\Mapping as ORM;
use Doctrine\DBAL\Types\Types;
use App\Entity\GsProjet\Project;
use App\Entity\GsProjet\User; // Ajout du bon namespace pour User

#[ORM\Entity(repositoryClass: MissionRepository::class)]
#[ORM\Table(name: "missions")]
#[ORM\HasLifecycleCallbacks]
class Mission
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 100)]
    private ?string $titre = null;

    #[ORM\Column(type: Types::TEXT, nullable: true)]
    private ?string $description = null;

    #[ORM\Column(length: 20)]
    private ?string $status = 'To Do';

    #[ORM\ManyToOne(targetEntity: Project::class, inversedBy: 'missions')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Project $project = null;
    #[ORM\ManyToOne(inversedBy: 'missions')]
    #[ORM\JoinColumn(name: 'assigned_to', referencedColumnName: 'id', nullable: true)]
    private ?User $assignedTo = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $createdAt = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE)]
    private ?\DateTimeInterface $updatedAt = null;

    #[ORM\Column(type: Types::DATETIME_MUTABLE, nullable: true)]
    private ?\DateTimeInterface $dateTerminer = null;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
        $this->updatedAt = new \DateTime();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): static
    {
        $this->titre = $titre;
        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): static
    {
        $this->description = $description;
        return $this;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(string $status): static
    {
        $this->status = $status;
        return $this;
    }

    public function getProject(): ?Project
    {
        return $this->project;
    }

    public function setProject(?Project $project): static
    {
        $this->project = $project;
        return $this;
    }

    public function getAssignedTo(): ?User
    {
        return $this->assignedTo;
    }

    public function setAssignedTo(?User $assignedTo): static
    {
        $this->assignedTo = $assignedTo;
        return $this;
    }

    public function getCreatedAt(): ?\DateTimeInterface
    {
        return $this->createdAt;
    }

    public function getUpdatedAt(): ?\DateTimeInterface
    {
        return $this->updatedAt;
    }

    public function getDateTerminer(): ?\DateTimeInterface
    {
        return $this->dateTerminer;
    }

    public function setDateTerminer(?\DateTimeInterface $dateTerminer): static
    {
        $this->dateTerminer = $dateTerminer;
        return $this;
    }
  
  
    public function setUpdatedAt(\DateTimeInterface $updatedAt): static
    {
        $this->updatedAt = $updatedAt;
        return $this;
    }
    #[ORM\PrePersist]
    #[ORM\PreUpdate]
    public function updateTimestamps(): void
    {
        $this->updatedAt = new \DateTime();
    }
    
    // Dans Mission.php



}
